drivers/jason0x43-wemo_switch.groovy

```diff
@@ -179,3 +188,84 @@ private debugLog(message) {
 private syncTime() {
     parent.childSyncTime(device)
 }
+
+
+def updatePort(port) {
+    iPort=s2i(port)
+    if (iPort < 1 || iPort > 65535 ) {
+        infoLog("Invalid TCP port specified - ${port}")
+        -1
+    } else {
+        hPort=i2h(iPort)
+        def existingPort = device.getDataValue('port')
+
+        infoLog("Existing Port: ${h2i(existingPort)}")
+        
+        if (iPort && iPort != h2i(existingPort)) {
+            debugLog("Updating port from ${h2i(existingPort)} to ${port}")
+            device.updateDataValue('port', hPort)
+        }
+        
+    }
+}
+
+def updateAddress(ip) {
+    
+    String[] splitIp = ip.split("\\.")
+    
+    if (splitIp.length !=4) {
+        infoLog("Invalid IP address specified - ${ip} (${splitIp.length} octets found)")
+        -1
+    } else {
+        intArrIp=strArrToIntArr(splitIp)
+        hexIp=intArrToHex(intArrIp)
+
+        def existingIp = device.getDataValue('ip')
+
+        infoLog("Existing IP: ${hexToIp(existingIp)}")
+        
+        if (ip && ip != hexToIp(existingIp)) {
+            debugLog("Updating IP from ${hexToIp(existingIp)} to ${ip}")
+            device.updateDataValue('ip', hexIp)
+        }
+    }
+}
+
+private strArrToIntArr(strArr) {
+    int[] intArr = new int[strArr.length];
+    for (int i = 0; i < strArr.length; i++) {
+        intArr[i]=s2i(strArr[i])
+    }
+    intArr
+}
+
+private h2i(hex) {
+    //Hex to Integer
+    Integer.parseInt(hex,16)
+}
+
+private s2i(strV) {
+    //String to Integer
+    Integer.parseInt(strV, 10)
+}
+
+private hexToIp(hex) {
+    //Hex IP String to dotted decimal IP string
+    [
+        h2i(hex[0..1]),
+        h2i(hex[2..3]),
+        h2i(hex[4..5]),
+        h2i(hex[6..7])
+    ].join('.')
+}
+
+private intArrToHex(arrAddress) {
```

Comment:
Rather than defining an `intArrToHex` function, import the HexUtils class (add `import hubitat.helper.HexUtils` after the `metadata` block) and just call `HexUtils.intArrayToHexString` directly.